### PR TITLE
GH-217: Resume training

### LINKED ISSUE #217: Resume training (keeping epoch, losses, and best model score)
It would be nice to be able to keep the epoch numbers and losses when resuming training on a model loaded from disk.

(https://pytorch.org/tutorials/beginner/saving_loading_models.html#saving-loading-a-general-checkpoint-for-inference-and-or-resuming-training)

#200 already mentioned keeping the scores in the pickled .pt object, but keeping the logs is also a nice add in (not overwriting them).

### PR DESCRIPTION
**Pull request by tabergma**
closes #217 

To be able to resume training you need to train with `checkpoint` set to `True`. 
If `checkpoint=True` the model plus the training parameters (e.g. loss, epoch, optimizer state dict) are saved to disk after every epoch. 
To continue training, for example for a `SequenceTagger`, you need to execute:
```
trainer = ModelTrainer.load_from_checkpoint('checkpoint.pt', 'SequenceTagger', corpus)
trainer.train(out_path)
```

### CODE REVIEW

**Comment by alanakbik** [comment]
:+1:
